GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using Microsoft.VisualStudio.Threading;
+using ResourceManager;
+
+namespace GitUI.UserControls
+{
+    public partial class InteractiveGitActionControl : GitModuleControl
+    {
+        private readonly TranslationString _progressMessage = new TranslationString("{0} is currently in progress.");
+        private readonly TranslationString _progressWithConflictsMessage = new TranslationString("{0} is currently in progress with merge conflicts.");
+
+        private readonly TranslationString _bisect = new TranslationString("Bisect");
+        private readonly TranslationString _rebase = new TranslationString("Rebase");
+        private readonly TranslationString _merge = new TranslationString("Merge");
+        private readonly TranslationString _patch = new TranslationString("Patch");
+
+        public enum State
+        {
+            None,
+            Bisect,
+            Rebase,
+            Merge,
+            Patch
+        }
+
+        private State _state = State.None;
+        private bool _hasConflicts = false;
+
+        public event EventHandler BisectEvent;
+
+        public InteractiveGitActionControl()
+        {
+            InitializeComponent();
+            InitializeComplete();
+        }
+
+        public async Task RefreshAsync()
```

Comment:
I don't really see any reasons for this method to be `async` - we are not doing any processing that should be on a background thread.
Switching between threads is not cheap.

All calls on `Module` instance are generally cheap - checking for presence of specific files. If for whatever reason we can't check these files (e.g. I/O operations take too long) than a user has a bigger issues to worry about than our UI not updating immediately.

Reply:
I was trying to avoid reintroducing the bug that was fixed by #2862. The issue there was calling InTheMiddleOfConflictedMerge on the UI thread, so an alternative would be to only use the background thread when calling that function?

Reply:
Actually, it seems that #2862 references the deprecated RunGitCmd function, so perhaps this is now fixed in a different way.

Reply:
I've removed all async functions and I couldn't repro the issue in #2862, but it might be worth someone who could see the issue originally taking a look.